aten/src/ATen/core/function_schema.h

```diff
@@ -145,7 +145,9 @@ struct FunctionSchema {
         arguments_(std::move(arguments)),
         returns_(std::move(returns)),
         is_vararg_(is_vararg),
-        is_varret_(is_varret) {}
+        is_varret_(is_varret) {
+    checkDefaultArgs();
```

Comment:
it would be nice if i could put it in the compiler but we do the following pattern currently:

```
  auto defined_functions = cu->define(
      QualifiedName(name.prefix()),
      {def},
      {pythonResolver(std::move(rcb))},
      nullptr,
      true);
  TORCH_INTERNAL_ASSERT(defined_functions.size() == 1);
  auto& defined = defined_functions[0];
  defined->setSchema(getSchemaWithNameAndDefaults(
      def.range(), defined->getSchema(), def.name().name(), defaults));
```
default arguments are added to the schema following compilation, the compiler doesn't have access to it